.github/workflows/dh-backend-cd.yml

```diff
@@ -11,49 +11,75 @@
-      - 'apps/dh/api-dh/**'
-      - '.github/workflows/dh-backend-ci.yml'
-      - '.github/actions/publish-release-contents/action.yml'
 
 jobs:
-  publish_release:
-    uses: Energinet-DataHub/.github/.github/workflows/promote-prerelease.yml@v8
-    with:
-      RELEASE_NAME_PREFIX: dh_backend
+  #
+  # Detect changes to start relevant jobs
+  #
 
-  dispatch_deployment_request:
-    needs: publish_release
-    uses: Energinet-DataHub/.github/.github/workflows/dispatch-deployment-request.yml@v8
-    with:
-      CALLER_REPOSITORY_NAME: greenforce-frontend
-      CALLER_REPOSITORY_PATH: Energinet-DataHub/greenforce-frontend
-      RELEASE_NAME_PREFIX: dh_backend
-      CUSTOM_EVENT_TYPE: dh-backend
-    secrets:
-      PAT_TOKEN: ${{ secrets.PAT_TOKEN }}
-      ENVIRONMENT_REPOSITORY_PATH: ${{ secrets.ENVIRONMENT_REPOSITORY_PATH }}
+  changes:
+    uses: ./.github/workflows/detect-changes.yml
+
+  #
+  # CD .NET
+  #
 
-  update_coverage_report:
+  dotnet_update_coverage_report:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
     uses: Energinet-DataHub/.github/.github/workflows/dotnet-solution-ci.yml@v8
     with:
       SOLUTION_FILE_PATH: 'apps/dh/api-dh/DataHub.WebApi.sln'
-      PREPARE_OUTPUTS: false
       ENVIRONMENT: AzureAuth
     secrets:
       AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
       AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
       AZURE_SPN_ID: ${{ secrets.AZURE_SPN_ID_OIDC }}
       AZURE_KEYVAULT_URL: ${{ secrets.AZURE_KEYVAULT_URL }}
       AZURE_SECRETS_KEYVAULT_URL: ${{ secrets.AZURE_SECRETS_KEYVAULT_URL }}
+
+  dotnet_promote_prerelease:
+    needs: changes
+    if: ${{ needs.changes.outputs.dotnet == 'true' }}
+    uses: Energinet-DataHub/.github/.github/workflows/promote-prerelease.yml@v8
+    with:
+      RELEASE_NAME_PREFIX: dotnet
+  #
+  # Dispatch deployment request
+  #
+
+  dispatch_deploment_event:
+    runs-on: ubuntu-latest
+    needs: [
+        # Necessary to be able to use its outputs in the dispatch event
+        changes,
+        # .NET
+        dotnet_promote_prerelease,
+      ]
+    if: |
+      always() &&
+      !contains(needs.*.result, 'failure') &&
+      !contains(needs.*.result, 'cancelled') &&
+      !(
+        needs.dotnet_promote_prerelease.result == 'skipped'
+      )
+    steps:
+      - run: echo "${{ toJSON(needs) }}"
+
+      - name: Find associated pull request
+        uses: Energinet-DataHub/.github/.github/actions/find-related-pr-number@v8
+        id: find_pull_request
+
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.PAT_TOKEN }}
+          repository: '${{ secrets.ENVIRONMENT_REPOSITORY_PATH }}'
+          event-type: 'geh-post-office-deployment-request-domain'
+          client-payload: '{"pr": "${{ steps.find_pull_request.outputs.pull_request_number }}", "dotnet_deploy": "${{ needs.changes.outputs.dotnet }}"}'
```

Comment:
In other domains we have changed the "pr" to "domain_pr".

But I'm not quite sure what the plan is here (in this repo), how many event types we expect and what should be able to trigger deployment.